framework/src/main/java/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -825,6 +892,16 @@ public Boolean visitTypevar_Typevar(
       } else if (!subtypeHasAnno && !supertypeHasAnno && areEqualInHierarchy(subtype, supertype)) {
         // two unannotated uses of the same type parameter are of the same type
         return true;
+      } else if (subtypeHasAnno) {
+        Set<AnnotationMirror> superLBs =
+            AnnotatedTypes.findEffectiveLowerBoundAnnotations(qualifierHierarchy, supertype);
+        AnnotationMirror superLB =
+            qualifierHierarchy.findAnnotationInHierarchy(superLBs, currentTop);
+        return qualifierHierarchy.isSubtype(subtype.getAnnotationInHierarchy(currentTop), superLB);
+      } else if (supertypeHasAnno) {
+        return qualifierHierarchy.isSubtype(
+            subtype.getEffectiveAnnotationInHierarchy(currentTop),
+            supertype.getAnnotationInHierarchy(currentTop));
 
       } else if (subtype.getUpperBound().getKind() == TypeKind.INTERSECTION) {
```

Comment:
The cascaded if statement is not exhaustive:  control can fall through.  Why is that?  It might be clearer to make the handling of the final case when `haveSameDeclaration` is true explicit, unless it needs all the following logic in the method.